"spack env update" can't handle override

When running:
```console
$ spack env update <env-dir>
```
if the `spack.yaml` file contains override with a double colon, the command will fail reporting that "Additional properties are not allowed".

### Steps to reproduce the issue
Create a `spack.yaml` file with the following content in a directory:
```yaml
spack:
  mirrors::
    test: /foo/bar
  packages:
    cmake:
      paths:
        cmake@3.18.1: /usr
  specs:
  - hdf5
```
Try then to run:
```console
$ spack env update <directory>
```

### Error Message
The command above fails with an error similar to:
```
[ ... ]
        try:
            spack.schema.Validator(schema).validate(test_data)
        except jsonschema.ValidationError as e:
            if hasattr(e.instance, 'lc'):
                line_number = e.instance.lc.line + 1
            else:
                line_number = None
>           raise ConfigFormatError(e, data, filename, line_number)
E           spack.config.ConfigFormatError: /tmp/pytest-of-culpo/pytest-0/test_can_update_attributes_wit0/spack.yaml: Additional properties are not allowed ('mirrors:' was unexpected)
```

### Information on your system

* **Spack:** 0.15.4-591-17d96b615
* **Python:** 3.7.4
* **Platform:** linux-ubuntu18.04-broadwell


### Additional information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have searched the issues of this repo and believe this is not a duplicate
- [ ] I have run the failing commands in debug mode and reported the output

--- a/lib/spack/spack/environment.py
+++ b/lib/spack/spack/environment.py
@@ -1787,7 +1787,7 @@ def update_yaml(manifest, backup_file):
 
     shutil.copy(manifest, backup_file)
     with open(manifest, 'w') as f:
-        _write_yaml(data, f)
+        syaml.dump_config(data, f)
     return True
 
 

--- a/lib/spack/spack/test/cmd/env.py
+++ b/lib/spack/spack/test/cmd/env.py
@@ -2112,3 +2112,23 @@ def test_old_format_cant_be_updated_implicitly(packages_yaml_v015):
     env('activate', str(manifest.dirname))
     with pytest.raises(spack.main.SpackCommandError):
         add('hdf5')
+
+
+@pytest.mark.regression('18147')
+def test_can_update_attributes_with_override(tmpdir):
+    spack_yaml = """
+spack:
+  mirrors::
+    test: /foo/bar
+  packages:
+    cmake:
+      paths:
+        cmake@3.18.1: /usr
+  specs:
+  - hdf5
+"""
+    abspath = tmpdir.join('spack.yaml')
+    abspath.write(spack_yaml)
+
+    # Check that an update does not raise
+    env('update', '-y', str(abspath.dirname))

